zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceElementTerminatedApplier.java

```diff
@@ -37,6 +41,10 @@ public void applyState(final long key, final ProcessInstanceRecord value) {
 
     bufferedStartMessageEventStateApplier.removeMessageLock(value);
 
+    if (value.getBpmnElementType() == BpmnElementType.MULTI_INSTANCE_BODY) {
+      multiInstanceState.deleteInputCollection(key);
```

Comment:
ℹ️ As deleting is backwards compatible (`deleteIfExists()`) I don't see the need of adding a new event applier version.